### DISCUSSION TITLE
[Suggestion] Improve Thumbnailer handler

### DISCUSSION THREAD

**Discussion by eduardomozart** [linuxdeepin/developer-center#2867] [Q&A | 问答板块]
Hello,

It's possible to register third-party thumbnailers into Nautilus, Nemo, Thunar and other files managers, but I could not found a way to register a thumbnail handler for non-default file types into DDE File Manager.

I see that DDE File Manager follows [Thumbnail Managing Standard](https://specifications.freedesktop.org/thumbnail-spec/thumbnail-spec-latest.html) from Freedesktop.org, but from far as I seen there is no standard way to register new thumbnailer extension/MIME type associations.

Some Debian packages creates a `.thumbnailer` file into `/usr/share/thumbnailer` (or `~/.local/share/thumbnailer`) directory for GNOME 3. E.g. "libgdk-pixbuf2.0-bin" package creates a `/usr/share/thumbnailer/librsvg.thumbnailer` that allows to create SVG thumbnails on-the-fly, but DDE File Manager do not seem to use this directory to handle thumbnail associations. There is a step-by-step on how to create thumbnails using this method into https://gubins.lv/posts/thumbnailer/

Into the past, GNOME 2 used a `gconf` approach to register custom thumbnailers (as seen in https://askubuntu.com/questions/16867/missing-thumbnails-for-swf-files-in-nautilus), but I do believe that Deepin do not have a similar approach to handle it's supported file thumbnailers.

It would be amazing if DDE File Manager supports to register custom thumbnailer for other MIME types instead of only support the built-in ones. I believe that following GNOME 3 standard is the best approach, as it seems to be the "standard" way to handle thumbnailers by other file managers, like Nautilus.

**Comment by zccrs** [comment]
The dde-file-manager supported use dynamic library plugin to improve thumbnailer and preview for files. @peeweep provide a example in there.
